src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/DefaultMonitorService.java

```diff
@@ -95,7 +101,12 @@ public MonitorConnectionContext startMonitoring(
 
   @Override
   public void stopMonitoring(MonitorConnectionContext context) {
-    final IMonitor monitor = this.threadContainer.getMonitorMap().get(context.getNodeKeys().iterator().next());
+    final String node = context.getNodeKeys()
```

Comment:
NIT: add a comment somewhere in this function to say that finding any node is good/enough for this function